provides a warning if profiles are read only

frickler24 opened the pull request:
This is a simple fix for issue #61. As @tatokis wrote, it is not a bug, but information for the user can help us minimize support effort.
The fix is based on master but AFAIK compatible to the other branches.

pourhadi committed:
Update main.c

add ability to randomly select color from gradient for type-lighting

(inspired by https://github.com/ccMSC/ckb/pull/264)

DefaultUser committed:
Add an init script for OpenRC

tatokis committed:
Add experimental support for K65 NRGB (0x1b07)

tatokis committed:
make debug now works with animations and inline removed from min/max in ckb-heat

tatokis committed:
ckb-heat: use fmin from math.h

tatokis committed:
ckb-heat: Usage count can not be negative. Fixes #30

tatokis committed:
Fix incorrect indentation from the previous commit.

ambadoom committed:
fix systemd automatic restarting

tatokis committed:
Added warning in readme that we don't autodetect OpenRC

tatokis committed:
Added Harpoon (0x1b3c) as a Sabre to see if it works.

tatokis committed:
usb.c: forgot to add == for comparison

tatokis committed:
Add K95 Platinum dev id (0x1b2d)

tatokis committed:
Remove urb count override that should have never been committed

frickler24 committed:
Fixes issue26: outdated FIRMWARE version

The file FIRMWARE in the root directory of ckb-next is signed.
Therefore, updating the obsolete content was not possible by anyone.

A new key has been created, with which only signatures for this development
should be made (ckb-next@frickler24.de, BAF07C6B).
  The key was self-signed and in addition signed with my personal key.
  It should also be signed by mattanger and possibly others.
Caution: The key may not be exported as ASCII file, but only as a binary file!

The display of gpg -d FIRMWARE should do something like this:
  Gpg: Signature from Sat 04 Feb 2017 11:59:11 CET using RSA key ID BAF07C6B
  Gpg: Correct signature of »ckb-next (For signing development artefacts only!)
       <Ckb-next@frickler24.de>«

To test the function, the signed FIRMWARE file was originally uploaded to a branch of mine:
 https://raw.githubusercontent.com/frickler24/ckb-next/issues-26-Firmware-Incident/FIRMWARE
 (in line 36, kbfirmware.cpp).

In order for the mechanism to go productive,
the FIRMWARE file of the MASTER branch must be updated!

To prepare the mechanism fpr production,
the link ist updated to the MASTER branch FIRMWARE file:
   https://raw.githubusercontent.com/mattanger/ckb-next/master/FIRMWARE

**********************
* Therefore this patch must first be used in master branch
* so that the correct file and key can be loaded at runtime.
* Then the patch can be added to the testing branch and others.
**********************

If the file or the key in the master branch is not up-to-date,
an error message is displayed when the ckb client is started
(as an example, here a possible error message):

  Gpg: Signature from Thu Jun 30th 2016 09:15:49 CEST using RSA key ID 8DC8D309
  Gpg: [do not know]: invalid packet (ctb = 2d)
  Gpg: keydb_search failed: Invalid package
  Gpg: Can not check signature: Public key not found

.gitignore is updated to ignore a directory containing tmp-files
and a script for signing the FIRMWARE-file.
If it is interesting for others, I can commit it also.

frickler24 committed:
added K70 variants

For the K70 the known variants are added.
The only one still missing is the K70 non rgb.

frickler24 committed:
added K70 Rapidfire RGB

There miight be an problem with this Firmware definition:
The filename at corsair doenload site is K70RGBRAPIDFIRE.
The standard rule used in ckb for creating the model names is
vendor + model-with-features + (RGB or nothing).
With this the name to look for in the FIRMWARE table should be
Corsair K70RAPIDFIRERGB

frickler24 committed:
added K95 Platinum RGB in FIRMWARE file

frickler24 committed:
change RAPIDFIRE to RFIRE

frickler24 committed:
renamed Firmware directory to .firmware

frickler24 committed:
provides a warning if profiles are read only

tatokis commented:
Unfortunately this doesn't seem to get triggered if the permissions in ~/.config/ckb (root:root) are set so that new files can't be created. In this case the lock.

```
tatokis@tatokis-PC:~/ckb-next/ckb-next-frickler24/bin$ strace -f -e open ./ckb 2>&1 | grep "\.config/ckb"
[pid 24592] open("/home/tatokis/.config/ckb/ckb.conf", O_RDONLY|O_CLOEXEC) = 17
[pid 24592] open("/home/tatokis/.config/ckb/ckb.conf", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 19
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 36
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf.lock", O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC, 0644) = -1 EACCES (Permission denied)
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 36
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf.lock", O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC, 0644) = -1 EACCES (Permission denied)
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 25
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf.lock", O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC, 0644) = -1 EACCES (Permission denied)
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 25
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf.lock", O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC, 0644 <unfinished ...>
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 25
[pid 24601] open("/home/tatokis/.config/ckb/ckb.conf.lock", O_WRONLY|O_CREAT|O_EXCL|O_CLOEXEC, 0644) = -1 EACCES (Permission denied)
```

mattanger committed:
This merge addresses issue #26

mattanger committed:
Replaced gpg key & FIRMWARE sig with ckb-next development team key

frickler24 committed:
Find more reasons for lack of readability of the config file

At the suggestion of @tatokis, not only the writability of the config
file itself is evaluated.
Now it searches also for reasons why the file could not be created.

This means that the standard function QSessions can no longer be used.
Attempts are first made to create a test entry in the Config.
If this test entry can be read after closing and reopening
the config file, the file is writable or can be created.

frickler24 committed:
just healing git trouble

frickler24 commented:
Sorry, I had some trouble with git again. I will  resend this PR after fixing it.

frickler24 commented:
New PR is #71
